### PR TITLE
[TAS-489] ✨ Add UI for discounted LIKE price

### PR DESCRIPTION
**Pull request by AuroraHuang22**
截圖 2024-01-16 下午5 54 14

### CODE REVIEW

File: src/mixins/nft.js [hunk 1]
```diff
@@ -313,12 +313,20 @@ export default {
...
+    nftPreDiscountedLIKE() {
+      return this.paymentInfo?.LIKEPricePrediscount;
+    },
     nftIsFree() {
       return this.NFTPrice !== undefined && this.NFTPrice === 0;
     },
     formattedNFTPriceInLIKE() {
       return this.NFTPrice ? formatNumberWithLIKE(this.nftPriceInLIKE) : '-';
     },
+    formattedNFTPreDiscountedLIKE() {
```
Comment (williamchong) [inline src/mixins/nft.js:325]:
```suggestion
    formattedNFTPrediscountedLIKE() {
```

File: src/mixins/nft.js [hunk 2]
```diff
@@ -313,12 +313,20 @@ export default {
     nftPriceInLIKE() {
       return this.paymentInfo?.LIKEPrice;
     },
+    nftPreDiscountedLIKE() {
```
Comment (williamchong) [inline src/mixins/nft.js:316]:
```suggestion
    nftPrediscountedLIKE() {
```

File: src/components/EventModal/Collect/index.vue [hunk 3]
```diff
@@ -660,6 +661,23 @@ export default {
...
+        return undefined;
+      }
+
+      const discountMultiplier = (discountedPrice / originalPrice) * 100;
+
+      return {
+        originalPriceInLIKE: this.formattedNFTPreDiscountedLIKE,
+        discountPercentage: discountMultiplier.toFixed(0),
+      };
+    },
```
Comment (williamchong) [inline src/components/EventModal/Collect/index.vue:680]:
```suggestion
    discountInfo() {
      if (!this.nftPrediscountedLIKE) {
        return undefined;
      }
      const originalPrice = this.nftPrediscountedLIKE;
      const discountedPrice = this.nftPriceInLIKE;
      if (originalPrice <= discountedPrice) {
        return undefined;
      }

      const discountMultiplier = (discountedPrice / originalPrice) * 100;

      return {
        originalPriceInLIKE: this.formattedNFTPrediscountedLIKE,
        discountPercentage: discountMultiplier.toFixed(0),
      };
    },
```

File: src/mixins/nft.js [hunk 4]
```diff
@@ -313,12 +313,20 @@ export default {
...
+    },
     nftIsFree() {
       return this.NFTPrice !== undefined && this.NFTPrice === 0;
     },
     formattedNFTPriceInLIKE() {
       return this.NFTPrice ? formatNumberWithLIKE(this.nftPriceInLIKE) : '-';
     },
+    formattedNFTPreDiscountedLIKE() {
+      return this.NFTPrice
+        ? formatNumberWithLIKE(this.nftPreDiscountedLIKE)
```
Comment (williamchong) [inline src/mixins/nft.js:327]:
```suggestion
        ? formatNumberWithLIKE(this.nftPrediscountedLIKE)
```

**Review by williamchong — CHANGES_REQUESTED** [review]
Let's treat prediscount is one word here and consistent with API field name

**Review by williamchong — APPROVED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**